### ISSUE TITLE
Add scene size check into editor to prevent hanging on displaying extra large scene trees

### ISSUE THREAD

**Issue by roalyr** [godotengine/godot-proposals#4817]
### Describe the project you are working on

GDTLancer. Generating a ton of star systems to fill up a galaxy, but wanting them to be pre-fabricated (partially).

### Describe the problem or limitation you are having in your project

I have generated a scene with 10000 nodes like this (I did it separately in Python, because it is faster):
```
[node name="BD-BD-300" type="Area" parent="." instance=ExtResource( 1 )]
transform = Transform( 1, 0, 0, 0, 1, 0, 0, 0, 1, -4.9004098565320237e+17, -9.134986206365277e+16, -3.006398086357045e+17)
script = ExtResource( 2 )

[node name="H-H-596" type="Area" parent="." instance=ExtResource( 1 )]
transform = Transform( 1, 0, 0, 0, 1, 0, 0, 0, 1, -4.6723416363435635e+17, -3.5029961960421212e+16, 5.965396601792234e+17)
script = ExtResource( 2 )

[node name="GML-GML-324" type="Area" parent="." instance=ExtResource( 1 )]
transform = Transform( 1, 0, 0, 0, 1, 0, 0, 0, 1, 4.326932671776246e+17, 9.472419151272832e+16, 3.240925028725789e+17)
script = ExtResource( 2 )

....
```
When loading the scene from within the script it all works fine, but if (accidentally) attempting to open a scene in the editor scene window it hands, while trying to display it.

Here is my scene file, for reference.
[Test_scene_tscn.txt](https://github.com/godotengine/godot-proposals/files/9050081/Test_scene_tscn.txt)
.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I have a few options:
1. Add pages or "show more" button to the scene tree window with limited number of top-level nodes to be displayed (on each unfolding?)

2. Add internal editor check on the number of scene nodes that have to be displayed and simply raise a warning against opening it in the editor (and show a code snippet to load it via code). This way it will prevent user from hanging the editor in such an edge case, but will allow to open the scene anyways too.

3. Add scene meta flag that will simply prevent it from being opened in editor. Such flag could be added manually to the .tscn file via text editor.

4. Suggest to open such scene in external text editor.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

1. Check scene node number, compare it to some value (could be set in options, say, 100?)
2. If there are more number than the limit - raise a warning when trying to display selected scene in the editor.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

The best work around is to not to open extra large scenes in the editor, but one may do that by accident.
In my case I would add "DO_NOT_CLICK" to the scene name and hope I don't.

### Is there a reason why this should be core and not an add-on in the asset library?

It is an editor feature.

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/pull/53679.

**Comment by roalyr** [comment]
> Related to [godotengine/godot#53679](https://github.com/godotengine/godot/pull/53679).

Also, is it possible to convert .tres to .res externally?

**Comment by Calinou** [comment]
> Also, is it possible to convert .tres to .res externally?

Yes, using the ResourceSaver class. However, there are no external tools for doing this, as `.res` and `.tres` are Godot-specific formats.
